include/NAS2D/FpsCounter.h

```diff
@@ -17,17 +17,18 @@ namespace NAS2D {
  *
  * FPS values are only approximates. As the FPS count gets higher, the returned value
  * becomes a more average count.
- *
- * \note	Because of the nature of FPS counters and their limited context, the
- *			FpsCounter has been designed with shared state. This means that multiple
- *			instances of FpsCounter will share values.
  */
 class FpsCounter
 {
 public:
-	FpsCounter();
-
 	unsigned int fps();
+
+private:
+	static constexpr unsigned int FpsCountsSize = 25;
```

Comment:
Consider making this `60` to compare the average across the minimally ideal 60 FPS every gamer strives for.

Adding min/max FPS as well is a useful metric.

Reply:
This is maybe starting to get out of scope with functional changes. This PR was meant to be  a refactor.

Though yes, I was considering changing that constant. It seemed a little arbitrary.

More long term, I'm questioning the calculation used for the FPS. It feels a bit inverted to me. Leeor wanted something stable that wouldn't jump around rapidly. Perhaps there are other ways to do the averaging to get a stable result. Maybe we can store the time delta, rather than an inverted FPS estimation based on the time for each individual frame.

Reply:
Milliseconds per frame is a good metric that most game devs use. Gamers on the other hand, only understand FPS.

I'm okay with either, just pick one and stick with it. :)

Reply:
I like knowing milliseconds per frame. Though yes, we should display FPS to end users.

Not sure yet on the internal representation to use. We can decide on that for future work. Probably look into the `std::chrono` stuff first.